aten/src/ATen/native/Resize.cpp

```diff
@@ -148,5 +186,88 @@ const Tensor& resize_(
   return self;
 }
 
+void resize_bytes_meta(StorageImpl* storage, c10::SymInt size_bytes) {
+  TORCH_CHECK(storage->resizable(), "Trying to resize storage that is not resizable");
+  storage->set_nbytes(size_bytes);
+}
+
+
+static void maybe_resize_storage_meta(TensorImpl* self, c10::SymInt new_size_bytes) {
+  // It does not make sense to try to resize a storage
+  // to hold 0 elements, and this can break
+  // if storage_offset is positive but
+  // new_size is 0, so just bail in that case
+  // (same comment is in Resize.h)
+  if (self->sym_numel() == 0) {
+    return;
+  }
+
+  const Storage& storage = self->unsafe_storage();
+  if (!storage) {
+    TORCH_INTERNAL_ASSERT(0, "NYI, this should only be Caffe2");
+  } else if (new_size_bytes > storage.nbytes()) {
+    resize_bytes_meta(storage.unsafeGetStorageImpl(), new_size_bytes);
+  }
+}
+
+inline TensorImpl* resize_impl_meta_(
+    TensorImpl* self,
+    c10::SymIntArrayRef size,
+    at::OptionalSymIntArrayRef stride,
+    bool resize_storage = true) {
+  if (self->sym_sizes() == size && (!stride || self->sym_strides() == stride.value())) {
+    return self;
+  }
+
+  const auto itemsize = self->dtype().itemsize();
+  const auto storage_offset = self->sym_storage_offset();
+  auto storage_size = c10::SymInt(1);
+  std::vector<SymInt> strides;
+  int64_t dim = size.size();
+  if (!stride.has_value()) {
+    strides.resize(dim);
+    // TODO: Move this into TensorImpl
```

Comment:
This needs to get refactored

Reply:
i see some resize changes were merged in #90442, @ezyang is there any more that needs to be done here?

Reply:
No, resize is all done!